gateway/handlers/deletefunction_proxy.go

```diff
@@ -0,0 +1,67 @@
+package handlers
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/openfaas/faas/gateway/metrics"
+	"github.com/openfaas/faas/gateway/requests"
+	"github.com/openfaas/faas/gateway/types"
+)
+
+// MakeDeleteFunctionProxyHandler creates a handler which forwards HTTP requests and sets replicasCount to 0
+func MakeDeleteFunctionProxyHandler(proxy *types.HTTPClientReverseProxy, notifiers []HTTPNotifier, baseURLResolver BaseURLResolver, metricsOptions metrics.MetricOptions) http.HandlerFunc {
+
+	return func(w http.ResponseWriter, r *http.Request) {
+
+		functionName, parseErr := getFunctionNameFromRequest(r)
+		if parseErr != nil {
+			log.Printf("Error parsing request to remove service.")
+		}
+
+		baseURL := baseURLResolver.Resolve(r)
+
+		requestURL := r.URL.Path
+
+		start := time.Now()
+
+		statusCode, err := forwardRequest(w, r, proxy.Client, baseURL, requestURL, proxy.Timeout)
```

Comment:
I've decided to reuse the previous functionality for DELETE as much as possible. So am reusing the `forwardRequest`